clients/core-java/src/main/java/org/commonjava/indy/client/core/IndyClientHttp.java

```diff
@@ -77,48 +81,58 @@
 
     private final String baseUrl;
 
-    private final URL url;
+    private final String apiVersion;
+
+    private List<Header> defaultHeaders;
 
     public IndyClientHttp( final IndyClientAuthenticator authenticator, final IndyObjectMapper mapper,
-                           SiteConfig location )
+                           SiteConfig location, String apiVersion )
             throws IndyClientException
     {
         this.objectMapper = mapper;
         this.location = location;
-
+        this.apiVersion = apiVersion;
         baseUrl = location.getUri();
-
-        try
-        {
-            url = new URL( baseUrl );
-        }
-        catch ( final MalformedURLException e )
-        {
-            throw new IndyClientException( "Invalid base-url: {}", e, baseUrl );
-        }
+        checkBaseUrl( baseUrl );
+        setDefaultHeaders();
 
         factory = new HttpFactory( authenticator );
     }
 
     public IndyClientHttp( final PasswordManager passwordManager, final IndyObjectMapper mapper,
-                           SiteConfig location )
+                           SiteConfig location, String apiVersion )
```

Comment:
Hmm, we should allow the user to set a list of default headers in the Indy client API constructor. This would allow the PNC and DA services to register their component ID for all comms to Indy. Combined with a build contentId, this would give us a pretty complete view of the traffic associated with a build.

Actually, we'd also have to allow each call to specify optional headers too...I'm not sure how to handle that yet, so let's table this for now.

Reply:
That might involve changes in ClientModules methods, Indy constructor, etc.